diff -Naur ./toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/bits/c++config.h /home/cwchung/toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/bits/c++config.h
--- ./toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/bits/c++config.h	2014-10-07 02:30:32.000000000 -0400
+++ /home/cwchung/toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/bits/c++config.h	2016-03-08 16:48:36.310392545 -0500
@@ -1341,7 +1341,7 @@
 /* #undef _GLIBCXX_X86_RDRAND */
 
 /* Define to 1 if mutex_timedlock is available. */
-#define _GTHREAD_USE_MUTEX_TIMEDLOCK 0
+#define _GTHREAD_USE_MUTEX_TIMEDLOCK 1
 
 #if defined (_GLIBCXX_HAVE__ACOSF) && ! defined (_GLIBCXX_HAVE_ACOSF)
 # define _GLIBCXX_HAVE_ACOSF 1
diff -Naur ./toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/thumb/bits/c++config.h /home/cwchung/toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/thumb/bits/c++config.h
--- ./toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/thumb/bits/c++config.h	2014-10-07 02:30:32.000000000 -0400
+++ /home/cwchung/toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/thumb/bits/c++config.h	2016-03-08 16:49:07.414393049 -0500
@@ -1341,7 +1341,7 @@
 /* #undef _GLIBCXX_X86_RDRAND */
 
 /* Define to 1 if mutex_timedlock is available. */
-#define _GTHREAD_USE_MUTEX_TIMEDLOCK 0
+#define _GTHREAD_USE_MUTEX_TIMEDLOCK 1
 
 #if defined (_GLIBCXX_HAVE__ACOSF) && ! defined (_GLIBCXX_HAVE_ACOSF)
 # define _GLIBCXX_HAVE_ACOSF 1
diff -Naur ./toolchain/include/c++/4.8/arm-linux-androideabi/bits/c++config.h /home/cwchung/toolchain/include/c++/4.8/arm-linux-androideabi/bits/c++config.h
--- ./toolchain/include/c++/4.8/arm-linux-androideabi/bits/c++config.h	2016-03-10 15:12:33.777100484 -0500
+++ /home/cwchung/toolchain/include/c++/4.8/arm-linux-androideabi/bits/c++config.h	2016-03-08 16:49:30.470393423 -0500
@@ -1341,7 +1341,7 @@
 /* #undef _GLIBCXX_X86_RDRAND */
 
 /* Define to 1 if mutex_timedlock is available. */
-#define _GTHREAD_USE_MUTEX_TIMEDLOCK 0
+#define _GTHREAD_USE_MUTEX_TIMEDLOCK 1
 
 #if defined (_GLIBCXX_HAVE__ACOSF) && ! defined (_GLIBCXX_HAVE_ACOSF)
 # define _GLIBCXX_HAVE_ACOSF 1
diff -Naur ./toolchain/include/c++/4.8/arm-linux-androideabi/thumb/bits/c++config.h /home/cwchung/toolchain/include/c++/4.8/arm-linux-androideabi/thumb/bits/c++config.h
--- ./toolchain/include/c++/4.8/arm-linux-androideabi/thumb/bits/c++config.h	2014-10-07 02:21:06.000000000 -0400
+++ /home/cwchung/toolchain/include/c++/4.8/arm-linux-androideabi/thumb/bits/c++config.h	2016-03-08 16:49:45.958393674 -0500
@@ -1341,7 +1341,7 @@
 /* #undef _GLIBCXX_X86_RDRAND */
 
 /* Define to 1 if mutex_timedlock is available. */
-#define _GTHREAD_USE_MUTEX_TIMEDLOCK 0
+#define _GTHREAD_USE_MUTEX_TIMEDLOCK 1
 
 #if defined (_GLIBCXX_HAVE__ACOSF) && ! defined (_GLIBCXX_HAVE_ACOSF)
 # define _GLIBCXX_HAVE_ACOSF 1
