--- a/toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/bits/c++config.h	2014-10-07 02:30:32.000000000 -0400
+++ b/toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/bits/c++config.h	2016-03-08 16:48:36.310392545 -0500
@@ -1341,7 +1341,7 @@
 /* #undef _GLIBCXX_X86_RDRAND */
 
 /* Define to 1 if mutex_timedlock is available. */
-#define _GTHREAD_USE_MUTEX_TIMEDLOCK 0
+#define _GTHREAD_USE_MUTEX_TIMEDLOCK 1
 
 #if defined (_GLIBCXX_HAVE__ACOSF) && ! defined (_GLIBCXX_HAVE_ACOSF)
 # define _GLIBCXX_HAVE_ACOSF 1
--- a/toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/bits/gthr-default.h	2016-03-14 12:45:40.008962752 -0400
+++ b/toolchain/include/c++/4.8/arm-linux-androideabi/armv7-a/bits/gthr-default.h	2016-03-14 12:44:06.820247754 -0400
@@ -46,6 +46,7 @@
 #endif
 
 #include <pthread.h>
+#include "android/timedlock.h"
 
 #if ((defined(_LIBOBJC) || defined(_LIBOBJC_WEAK)) \
      || !defined(_GTHREAD_USE_MUTEX_TIMEDLOCK))
